AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -5265,79 +5247,27 @@ function Compress-AzResourceId {
+    Correctly import the MsGraph module for your PowerShell version and then sign in using the same tenant is the currently signed in Az user. This wrapper is necessary as 1. AzureAD is not directly compatible with PowerShell 6 (though this can be achieved through the WindowsCompatibility module), and 2. AzureAD doesn't necessarily log you into the same tenant as the Az cmdlets according to their documentation (although it's not clear when it doesn't).
     .EXAMPLE
-    Request-ConnectAzureAD
-    #>
-
-    [CmdletBinding()]
-    param()
-
-    Assert-IsWindows
-    Request-AzureADModule
-
-    $aadModule = Get-Module | Where-Object { $_.Name -like "AzureAD" }
-    if ($null -eq $aadModule) {
-        if ($PSVersionTable.PSVersion -ge [Version]::new(6,0,0,0)) {
-            Import-WinModule -Name AzureAD -Verbose:$false
-        } else {
-            Import-Module -Name AzureAD
-        }
-    }
-
-    try {
-        Get-AzureADTenantDetail -ErrorAction Stop | Out-Null
-    } catch {
-        $context = Get-AzContext
-        Connect-AzureAD `
-                -TenantId $context.Tenant.Id `
-                -AccountId $context.Account.Id `
-                -AzureEnvironmentName $context.Environment.Name | `
-            Out-Null
-    }
-}
-
-function Get-AzureADDomainInternal {
-    <#
-    .SYNOPSIS
-    Get the Azure AD domains associated with this Azure AD tenant.
-    .DESCRIPTION
-    This cmdlet is a wrapper around Get-AzureADDomain that is provided to future proof for adding cross-platform support, as AzureAD is not a cross-platform PowerShell module.
-    .PARAMETER Name
-    Specifies the name of a domain.
-    .EXAMPLE
-    $domains = Get-AzureADDomainInternal
-    .EXAMPLE
-    $specificDomain = Get-AzureADDomainInternal -Name "contoso.com"
-    .OUTPUTS
-    Microsoft.Open.AzureAD.Model.Domain
-    Deserialized.Microsoft.Open.AzureAD.Model.Domain, if accessed through the WindowsCompatibility module
+    Request-ConnectMsGraph
     #>
 
     [CmdletBinding()]
     param(
-        [Parameter(Mandatory=$false, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)]
-        [string]$Name
+        [Parameter(Mandatory=$true)]
+        [string[]]$Scopes
     )
 
-    begin {
-        Assert-IsWindows
-        Request-ConnectAzureAD
-    }
-
-    process {
-        $getParams = @{}
-        if ($PSBoundParameters.ContainsKey("Name")) {
-            $getParams += @{ "Name" = $Name}
-        }
+    Assert-IsWindows
+    Request-MSGraphModule
 
-        return (Get-AzureADDomain @Name)
-    }
+    $context = Get-AzContext
+    Connect-MgGraph -Scopes $Scopes -TenantId $context.Tenant.Id | Out-Null

```

Comment:
Seems like we used to connect to a particular account ID as well, with `-AccountId $context.Account.Id`. Is that not possible anymore in MS Graph? Or are there no advantages to doing so?

Reply:
The MS graph does not have an account parameter. There is no pre-selected account that needs to be passed.
By passing Connect-MgGraph you will get an Authorization pop-up through which you can login with user credentials.

Reply:
Nice!